### DISCUSSION TITLE
Run action on repo creation from template

### DISCUSSION THREAD

**Discussion by themoch** [community/community#25748] [Actions]
<p>I would like to generate scoped NodeJs repo from a template.</p>
<p>I have the template wired up with an action that already has the ability to run and create the package.json, then commit that to the repo.  What event can I trigger the action on so that when somebody clicks the “use this template” it will run the action that finishes the setup?</p>
<p>I’ve tried to use: </p>
<pre><code>release:
    types: # This configuration does not affect the page_build event above
      - created
</code></pre>
<p>but that doesn’t seem to be the correct workflow.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/themoch">@themoch</a> , </p>
<p>I would recommend you using <a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#create-event-create" rel="nofollow noopener">create event</a>.   </p>
<p>Add on: create to the workflow yml which could create package.json in your template repo . </p>
<pre><code>on:
  create:
</code></pre>
<p>After done this and clicking on “use the template”, a new repo will be created . And a workflow run on: create will be triggered. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/b/bbd072ff2b200b9937dd31947f6ce2cb48570ef2.png" data-download-href="/uploads/short-url/qNtT0VyidaIy013qEYmFj4dbemK.png?dl=1" title="create event.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181093022-c86764cc-0e83-420b-b465-a9f3475b48eb.png" alt="create event.png" data-base62-sha1="qNtT0VyidaIy013qEYmFj4dbemK" width="690" height="317" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/b/bbd072ff2b200b9937dd31947f6ce2cb48570ef2_2_690x317.png, https://aws1.discourse-cdn.com/github/original/2X/b/bbd072ff2b200b9937dd31947f6ce2cb48570ef2.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/b/bbd072ff2b200b9937dd31947f6ce2cb48570ef2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/b/bbd072ff2b200b9937dd31947f6ce2cb48570ef2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">create event.png</span><span class="informations">898×413 17.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Please kindly let me know whether this could work for you . </p>

**Comment by themoch** [comment]
<p>It worked! </p>
<p>I’m looking over the other documentation about events for Actions (<a href="https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes" rel="nofollow noopener">https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onevent_nametypes</a>) and the value they give is <code>created</code> not <code>create</code>. It does not work with <code>created</code>.  At least the autosuggest in the Actions Editor on Github shows <code>create</code>.</p>
<p>Thank you for your reply.</p>

**Comment by Yanjingzhu** [comment]
<p>Hi <a class="mention" href="/themoch">@themoch</a> , </p>
<p>I want to provide some explanation. </p>
<p>The doc you mentioned before links to  <strong>on.&lt;event_name&gt;.types </strong> , the “created” is the type name of release event. All the Activity types of events are in past tense. </p>
<p>I have recommended you to use <strong>create</strong> event, the “create” is the name of an event. </p>

**Comment by themoch** [comment]
<p>I understand completely.</p>
<p>It would be nice if this page contained a list of events at the top or in the navigation for quick reference.  </p>
<p><a href="https://help.github.com/en/actions/reference/events-that-trigger-workflows#create-event-create" rel="nofollow noopener">https://help.github.com/en/actions/reference/events-that-trigger-workflows#create-event-create</a></p>

**Comment by Yanjingzhu** [comment]
<p>Oh, there is a good idea. We will consider it. </p>

**Comment by timtribers** [comment]
<p>Hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a>, is this still accurate advice?</p>
<p>I have:<br>
<code>on: [create, workflow_dispatch]</code><br>
in an initialise.yml action in my Template.</p>
<p>But when a user creates a repo from the template, it doesn’t run :frowning_face:<br>
Running the workflow manually works fine.</p>
<p>Any thoughts?</p>

**Comment by mikaeldui** [comment]
<p>Doesn’t work for me either.</p>
<p>As a work-around, one could have <code>on: push</code> instead in the template and then edit it out and commit the new modified initialise.yml, if one wants to keep it. Personally I just delete the workflow as the last step before committing the changes.</p>
<p>Maybe with an <code>if: ${{ github.repository != 'owner/template-repo' }}</code>.</p>

**Comment by r-chaykovskiy** [comment]
To me it's like there's a 50/50 chance that the `create` event will fire. Sometimes it works, sometimes it doesn't. Very inconsistent and buggy....
